### PR TITLE
chore: Fix the PullQueryRoutingFunctionalTest

### PR DESCRIPTION
**Pull request by jnh5y**

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/ClusterStatusResource.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ private HostStoreLags getHostStoreLags(final KsqlHostInfo ksqlHostInfo, final bo
     final Map<String, ActiveStandbyEntity> perQueryMap = new HashMap<>();
     for (PersistentQueryMetadata persistentQueryMetadata: engine.getPersistentQueries()) {
       for (StreamsMetadata streamsMetadata: persistentQueryMetadata.getAllMetadata()) {
-        if (streamsMetadata.hostInfo().equals(asHostInfo(ksqlHostInfo))) {
+        if (!streamsMetadata.hostInfo().equals(asHostInfo(ksqlHostInfo))) {
```
Comment (jnh5y) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/ClusterStatusResource.java:105]: https://github.com/confluentinc/ksql/commit/9615a0296bfacd6eea9f054958319987792865f4#diff-e7ce1cd863f76d6f652a7345d1289367dbf96a963cdd63afe2c192c1c0c5a944L106-L107 dropped the negation in front of `streamsMetadata.hostInfo().equals(asHostInfo(ksqlHostInfo)`.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/DiscoverRemoteHostsUtil.java [hunk 2]
```diff
@@ -42,6 +42,7 @@ private DiscoverRemoteHostsUtil() {
         .map(QueryMetadata::getAllMetadata)
         .filter(Objects::nonNull)
         .flatMap(Collection::stream)
+        .filter(streamsMetadata -> !streamsMetadata.hostInfo().equals(HostInfo.unavailable()))
```
Comment (jnh5y) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/DiscoverRemoteHostsUtil.java:45]: This seems to capture the filtering check from here: https://github.com/confluentinc/ksql/commit/9615a0296bfacd6eea9f054958319987792865f4#diff-b5d09835b101449b87f099018515be248e21a66ffb0affae62041070ccd7ff55L47

**Comment by jnh5y** [comment]
This PR (or another) may need to add a check to `AllHostsLocator` (https://github.com/confluentinc/ksql/commit/9615a0296bfacd6eea9f054958319987792865f4#diff-b80651600282940dcd08cde116887ec029e053bab151ea740523de7cb214f0d6L65) like the other check for unavailable HostInfo.

**Review by stevenpyzhang — APPROVED** [review]
lgtm

**Pull request merged**